src/main/java/Scruffles.java

```diff
@@ -0,0 +1,225 @@
+                break;
+            } else if (input.equals("list")) {
+                for (int i = 0; i < list.size(); i++) {
+                    String output = (i + 1) + "." + list.get(i).toString();
+                    System.out.println(output);
+                }
+                if (list.isEmpty()) {
+                    System.out.println("you have no tasks woof woof!");
+                }
+            } else if (input.startsWith("mark")) {
+                try {
+                    if (input.equals("mark") || input.equals("mark ")) {
+                        throw new DescriptionEmptyException("grrrr >:( you need to mark something woof woof!");
+                    } else {
+                        input = input.replaceFirst("mark ", "");
+                        int listRef = Integer.parseInt(input) - 1;
+                        if (listRef >= list.size() || listRef < 0) {
+                            throw new OutOfBoundsException(listRef + 1);
+                        } else {
+                            list.get(listRef).setDone();
+                            System.out.println("woof! the task is now marked as done woof!");
+                            System.out.println(list.get(listRef).toString());
+                        }
+                    }
+                } catch (NumberFormatException e) {
+                    System.out.println("grrrr >:( you need to input an integer woof woof!");
+                } catch (DescriptionEmptyException | OutOfBoundsException e) {
+                    System.out.println(e.getMessage());
+                }
+            } else if (input.startsWith("todo")) {
+                try {
+                    if (input.equals("todo") || input.equals("todo ")) {
+                        throw new DescriptionEmptyException();
+                    } else {
+                        input = input.replaceFirst("todo ", "");
+                        list.add(new Todo(input));
+
+                        System.out.println("woof! the task is added woof!");
+                        System.out.println(list.get(taskCount).toString());
+                        System.out.println("you now have " + (taskCount + 1) + " tasks in the list woof!");
+
+                        taskCount++;
+                    }
+                } catch (DescriptionEmptyException e) {
+                    System.out.println(e.getMessage());
+                }
+            } else if (input.startsWith("deadline")) {
+                try {
+                    if (input.equals("deadline") || input.equals("deadline ")) {
+                        throw new DescriptionEmptyException();
+                    } else if (!input.contains(" /by ")) {
+                        throw new DescriptionEmptyException("grrrr >:( when is your deadline?? woof woof!");
+                    } else {
+                        input = input.replaceFirst("deadline ", "");
+                        String[] inputArray = input.split(" /by ");
```

Comment:
Perhaps plural form should be used on names representing a collection of objects?
```suggestion
                        String[] inputs = input.split(" /by ");
```